ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import {TagsInput} from '../../../shared/components/tags-input/tags-input';
+import {NamespaceFilter} from '../../../shared/components/namespace-filter';
+import {CheckboxFilter} from '../../../shared/components/checkbox-filter/checkbox-filter';
+import * as models from '../../../../models';
+
+require('./workflow-filters.scss');
+
+function getLabelsSuggestions(labels: Map<string, Set<string>>) {
+    const suggestions = new Array<string>();
+    Array.from(labels.entries()).forEach(([label, values]) => {
+        values.forEach(val => suggestions.push(`${label}=${val}`));
+    });
+    return suggestions;
+}
+
+interface WorkflowFilterProps {
+    workflows: models.Workflow[];
+    namespace: string;
+    phases: string[];
+    labels: string[];
+    onChange: (namespace: string, phases: string[], labels: string[]) => void;
+}
+
+interface WorkflowFilterState {
+    namespace: string;
+    phases: string[];
+    labels: string[];
+    error?: Error;
+}
+
+export class WorkflowFilters extends React.Component<WorkflowFilterProps, WorkflowFilterState> {
+    constructor(props: Readonly<WorkflowFilterProps>) {
+        super(props);
+        this.state = {
+            namespace: props.namespace,
+            phases: props.phases,
+            labels: props.labels
+        };
+    }
+
+    private set namespace(namespace: string) {
+        this.setState({namespace});
+    }
+
+    private get namespace() {
+        return this.state.namespace;
+    }
+
+    private set phases(phases: string[]) {
+        this.setState({phases});
+    }
+
+    private get phases() {
+        return this.state.phases;
+    }
+
+    private set labels(labels: string[]) {
+        this.setState({labels});
+    }
+
+    private get labels() {
+        return this.state.labels;
+    }
+
+    render() {
+        const {workflows, onChange} = this.props;
+        const phasesMap = new Map<string, number>();
+        Object.values(models.NODE_PHASE).forEach(value => phasesMap.set(value, 0));
```

Comment:
Using explicitly defined Maps like `phasesMap` and `labelsMap` smells like an anti-pattern in JS/TS. Let's redesign these in person on Monday.